### PR TITLE
Initial blockly reference contribution

### PR DESCRIPTION
**Pull request by stefan-hoehn**
Initial pull request to transfer the whole blockly reference to the official openHAB docs.

@Confectrician please have a review and get in touch with me

### CODE REVIEW

**Comment by Confectrician** [comment]
Hi Stefan,

Before having a closer look.
Does the complete reference apply to openHAB 3.2.0 already?

We could then also add this to the stable docs, when it is finished.

File: configuration/rules-dsl.md [hunk 1]
```diff
@@ -7,6 +7,8 @@ title: Rules
 
 "Rules" are used for automating processes: Each rule can be triggered, which invokes a script that performs any kinds of tasks, e.g. turn on lights by modifying your items, do mathematical calculations, start timers etcetera.
 
+Note that there is also a visual way of programming openHAB rules, which may be more suitable for beginners. Its documentation can be found in the [Blockly Reference section]({{base}}/configuration/rules-blockly.html)
```
Comment (Confectrician) [inline configuration/rules-dsl.md:10]: Please change this to one sentence per line.
The website build will take core of a proper presentation of paragraphs.

File: configuration/rules-blockly.md [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+[![Timers and Delays](images/blockly/blockly-date-handling-ephemeris-small.png "Ephemeris Dates")
+](rules-blockly-date-handling.html) [![Timers and Delays](images/blockly/blockly-date-handling-persistence-small.png "Persistence Dates")
+](rules-blockly-date-handling.html)
+
+See [Date Handling](rules-blockly-date-handling.html) section.
+
+### Ephemeris
+
+The ephemeris category provides blocks with calendar functionality. The blocks can be used to determine what type of day today is, or a number of days before or after today is. For example, a way to determine if today is a weekend, a bank holiday, someone’s birthday, trash day, etc.
```
Comment (Confectrician) [inline configuration/rules-blockly.md:79]: Please change this to one sentence per line.
The website build will take core of a proper presentation of paragraphs.

File: configuration/rules-blockly.md [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+
+[![Timers and Delays](images/blockly/blockly-timers-and-delays-small.png "Items and Things")
+](rules-blockly-timers-and-delays.html)
+
+See [Timers and Delays](rules-blockly-timers-and-delays.html) section.
+
+### Date Handling
+
+Date blocks are used as input parameters for other blocks.
+At the moment some of these blocks are found in the ephemeris section, whilst others are found in the persistence section - they may be equally useful for both. These blocks are *typed* to assure correct connection to other blocks.
```
Comment (Confectrician) [inline configuration/rules-blockly.md:69]: Please change this to one sentence per line.
The website build will take core of a proper presentation of paragraphs.

File: configuration/rules-blockly.md [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+*Items* and *Things* are the [major entities of openHAB](https://www.openhab.org/docs/concepts/) to control and monitor the home.
+
+[![Items and Things](images/blockly/blockly-items-and-things-small.png "Items and Things")
+](rules-blockly-items-things.html)
+
+See [Items & Things](rules-blockly-items-things.html) section
+
+### Timers and Delays
+
+Timers and Delays are a little more complex but add important functionality to rules. Whilst the “Wait-For”-block is straightforward, the timer blocks should be well understood before being used - they may behave differently than expected. This chapter explains what these blocks do, sometimes displaying generated code to explain what is happening behind the scenes.
```
Comment (Confectrician) [inline configuration/rules-blockly.md:58]: Please change this to one sentence per line.
The website build will take core of a proper presentation of paragraphs.

File: configuration/rules-blockly.md [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+
+This section provides a detailed description of the specific blocks and provides examples on how to use them. Note that some of the blocks (like voice, streaming or notifications) need some special setup within openHAB  - in these case links to the respective documentation is provided. It should also be mentioned that each of the blocks do have a context sensitive menu which appears upon a right click on the block itself.
+
+*Block context menu*
+
+![blockly-context.png](images/blockly/blockly-contextmenu.png)
+
+**Looking for help**
+
+A special mention should go towards the **Help** entry that links to a resource that is usually very helpful to understand the context of that particular block. Please read this information first before asking questions in the forum. *In case you ask for help please always post the respective code that is being generated.*
```
Comment (Confectrician) [inline configuration/rules-blockly.md:39]: Please change this to one sentence per line.
The website build will take core of a proper presentation of paragraphs.

File: configuration/rules-blockly.md [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+*Blockly toolbox*
+
+![blockly-toolbox-1](images/blockly/blockly-toolbox-1.png)![blockly-toolbox-2](images/blockly/blockly-toolbox-2.png)![blockly-toolbox-3](images/blockly/blockly-toolbox-3.png)
+
+
+
+All of these provide general functionality that is not specific to openHAB itself. If you want to learn more about how to use them, search for the many blockly tutorials that are available.
+However, to leverage the full capabilities more than *50 specific blocks* have been provided that are tailored for easy access of openHAB's capabilities.
+
+This section provides a detailed description of the specific blocks and provides examples on how to use them. Note that some of the blocks (like voice, streaming or notifications) need some special setup within openHAB  - in these case links to the respective documentation is provided. It should also be mentioned that each of the blocks do have a context sensitive menu which appears upon a right click on the block itself.
```
Comment (Confectrician) [inline configuration/rules-blockly.md:31]: Please change this to one sentence per line.
The website build will take core of a proper presentation of paragraphs.

File: configuration/rules-blockly.md [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+The basic idea behind the visual paradigm and representation within openHAB is based on the [Google Blockly Support](https://developers.google.com/blockly) which has been integrated and which provides the basic blocks for programming like the ones on the left and the right side of the below images
+
+{: #blockly-toolbox}
+*Blockly toolbox*
+
+![blockly-toolbox-1](images/blockly/blockly-toolbox-1.png)![blockly-toolbox-2](images/blockly/blockly-toolbox-2.png)![blockly-toolbox-3](images/blockly/blockly-toolbox-3.png)
+
+
+
+All of these provide general functionality that is not specific to openHAB itself. If you want to learn more about how to use them, search for the many blockly tutorials that are available.
```
Comment (Confectrician) [inline configuration/rules-blockly.md:28]: Please change this to one sentence per line.
The website build will take core of a proper presentation of paragraphs.

File: configuration/rules-blockly.md [hunk 8]
```diff
@@ -0,0 +1,157 @@
+---
+layout: documentation
+title: Rules Blockly
+---
+
+# Blockly Reference
+
+One of the core feature that OpenHAB provides is writing rules to allow specific behaviour with the home automation system. The usual way of developing rules is by coding them like described in the [Textual Rules]({{base}}/configuration/rules-dsl.html). However, this art of programming may become intimidating early on and shy away away people with few or almost no experience in programming. Therefore openHAB also provides a graphical way of writing rules which allows to put together rules in a rather visual way (even though some programming background may still help).
```
Comment (Confectrician) [inline configuration/rules-blockly.md:8]: Please change this to one sentence per line.
The website build will take core of a proper presentation of paragraphs.

File: configuration/rules-blockly-voice-and-multimedia.md [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+*Function:* Plays an audio file on an audio sink and setting the volume at the same time
+
+same as above but also allowing to provide the volume at the same time
+
+
+- the audio file must reside in the sounds-folder of openHAB that can be found [here](https://community.openhab.org/t/blockly-reference/128785#openhab-configuration-files-7)
+- all available audio sinks are automatically provided by openHABs blocklies
+
+**Known issue workaround:**
+In several cases (like the google minis) it is known that setting the volume together with playing the sound results into only setting the volume either during playing the sound or even after the sound has played. In case this happens the more robust approach is as follows
```
Comment (Confectrician) [inline configuration/rules-blockly-voice-and-multimedia.md:63]: Please change this to one sentence per line.
The website build will take core of a proper presentation of paragraphs.

File: configuration/rules-blockly-voice-and-multimedia.md [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+
+## Setup
+
+* Audio Sinks
+  - Audio devices are automatically added by their bindings as *sinks* in openHAB
+  -  [Reference on how to setup Audio and Audio Sinks](https://www.openhab.org/docs/configuration/multimedia.html)
+* Text-To-Speech
+  - In order to use text-to-speech, you need to install at least one [TTS service](https://www.openhab.org/addons/#voice).
+  - see [openHAB Voice](https://www.openhab.org/docs/configuration/multimedia.html#voice) for more information
+  - in case you only have one TTS service, this becomes your default TTS services. In case you have several TTS services installed, you have to configure one of them as the default which can be done in your [services.cfg ](https://github.com/openhab/openhab-distro/blob/master/distributions/distribution-resources/src/main/resources/runtime/etc/services.cfg#L20). See more on that special case [here](https://community.openhab.org/t/where-to-set-default-tts-voice/15450/2)
```
Comment (Confectrician) [inline configuration/rules-blockly-voice-and-multimedia.md:37]: Please change this to one sentence per line.
The website build will take core of a proper presentation of paragraphs.

Should also work foir list elements when you do not add the next dash for the next list element

File: configuration/rules-blockly-voice-and-multimedia.md [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+[return to Blockly Reference]({{base}}/configuration/rules-blockly.html)
+
+## Introduction
+
+In general, this section deals with two topics
+
+1) Playing or streaming audio to an audio sink e.g a speaker
+2)  Saying a text via using any Text-to-Speech API (e.g. Google's API)
+
+It should be noted that both functionalities do not just work right out of the box because e.g for (1) the audio sinks need to be installed in openHAB and for (2) e.g. the whole Google API needs to be configured in particular with your own personal google account. Even though (2) is not trivial but after having done that the usage of the functionality with blockly is very easy and worth the effort.
```
Comment (Confectrician) [inline configuration/rules-blockly-voice-and-multimedia.md:16]: Please change this to one sentence per line.
The website build will take core of a proper presentation of paragraphs.

File: configuration/rules-blockly-value-storage.md [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+layout: documentation
+title: Rules Blockly - Value Storage
+---
+
+# Value Storage
+[return to Blockly Reference]({{base}}/configuration/rules-blockly.html)
+
+## Introduction
+
+These blocks enable storing information *for a rule* that is kept after the rule has run, so it can be reused when the rule is run again later in stateful way. Basically a value storage can be perceived as a global variable for the rule instance.
```
Comment (Confectrician) [inline configuration/rules-blockly-value-storage.md:11]: Please change this to one sentence per line.
The website build will take core of a proper presentation of paragraphs.

File: configuration/rules-blockly-timers-and-delays.md [hunk 13]
```diff
@@ -0,0 +1,255 @@
...
+*Function*: returns `true` if the code *within the timer block* is currently running - i.e. the countdown has finished, and the code within the timer block is currently executing. This event will usually only last a few milliseconds, and you’d be (un)lucky to catch it when your rule retriggers.
+
+### Timer has terminated
+
+![timer-terminated](images/blockly/blockly-timer-terminated.png)
+*Function*: returns `true` if the timer has finished its countdown, and the code within the timer block has finished running.
+
+### Comprehensive Timer Example
+
+The intent here is to trigger an alarm to the police but allow the timer to be reset in case a reset button has been pressed during the trigger time. The rule is triggered when the burglar-alarm-item or the reset-button-item is changed.
```
Comment (Confectrician) [inline configuration/rules-blockly-timers-and-delays.md:182]: Please change this to one sentence per line.
The website build will take core of a proper presentation of paragraphs.

File: configuration/rules-blockly-timers-and-delays.md [hunk 14]
```diff
@@ -0,0 +1,255 @@
...
+In the case of *do nothing* the `else{}` branch is empty (which turns to be almost equals to the simle-timer).
+```javascript
+if (typeof this.timers['nothingTimerBlock'] === 'undefined' || this.timers['nothingTimerBlock'].hasTerminated()) {
+  this.timers['nothingTimerBlock'] = scriptExecution.createTimer(zdt.now().plusSeconds(10), function () {
+    })
+} else {
+  // do nothing
+}
+```
+In the case of *cancel* the `else{}` branch contains code to cancel the timer.
```
Comment (Confectrician) [inline configuration/rules-blockly-timers-and-delays.md:132]: Please ad an empty line after the code block befor starting the next paragraph.

File: configuration/rules-blockly-timers-and-delays.md [hunk 15]
```diff
@@ -0,0 +1,255 @@
...
+  this.timers['MyTimer'].reschedule(zdt.now().plusSeconds(10));
+}
+```
+
+**Simple timer-block**
+
+The simple timer-block generates the code shown underneath the image below.
+
+![simple-timer](images/blockly/blockly-simple-timer.png)
+```javascript
```
Comment (Confectrician) [inline configuration/rules-blockly-timers-and-delays.md:109]: Please add an empty line before the code block.

File: configuration/rules-blockly-timers-and-delays.md [hunk 16]
```diff
@@ -0,0 +1,255 @@
...
+![afterperiodtimer-example](images/blockly/blockly-afterperiod-timer-example.png)
+
+### Reschedule Timer
+
+*Function:*  Sometimes one may want to use the same timer and reschedule the same block of code again. This can be achieved with the reschedule-block
+
+![reschedule-timer](images/blockly/blockly-reschedule-timer.png)
+
+An example is as follows:
+* The code within the timer-block will run after 5 seconds.
```
Comment (Confectrician) [inline configuration/rules-blockly-timers-and-delays.md:64]: Please use `-` for lists in all files.

Currently we have mixed usage of `-` and `*`.

File: configuration/rules-blockly-timers-and-delays.md [hunk 17]
```diff
@@ -0,0 +1,255 @@
...
+![waifor-example](images/blockly/blockly-waitfor-example.png)
+
+### After *period of time* Do With Timer
+
+![blockly-afterperiodtimer](images/blockly/blockly-afterperiod-timer.png)
+
+*Function:* Schedule a task to be performed once after a specific amount of time has elapsed.
+
+- A number and unit (seconds, minutes up to months) can be chosen after which the commands within the block will be executed
+- Provide the name of the timer, allowing further access to that timer within the same rule. The name also ensures the timer is unique within the rule.
```
Comment (Confectrician) [inline configuration/rules-blockly-timers-and-delays.md:49]: Please change this to one sentence per line.
The website build will take care of a proper presentation of paragraphs.

File: configuration/rules-blockly-timers-and-delays.md [hunk 18]
```diff
@@ -0,0 +1,255 @@
...
+
+## Overview of the Timers and Delays category
+
+> ![timers-and-delays](images/blockly/blockly-timers-and-delays-1.png) ![timers-and-delays.png](images/blockly/blockly-timers-and-delays-2.png)
+
+##Timer Naming
+
+Timers are created and referred to by name, enabling manipulation within a rule.
+
+> **Important**: a named timer is *only* available within the same rule. The same timer *cannot* be accessed via a different rule. A different rule with a timer of the same name results into two separate timers.
```
Comment (Confectrician) [inline configuration/rules-blockly-timers-and-delays.md:26]: Please change this to one sentence per line.
The website build will take core of a proper presentation of paragraphs.

File: configuration/rules-blockly-timers-and-delays.md [hunk 19]
```diff
@@ -0,0 +1,255 @@
+---
+layout: documentation
+title: Rules Blockly - Timers & Delays
+---
+
+# Timers & Delays
+[return to Blockly Reference]({{base}}/configuration/rules-blockly.html)
+
+Timers and Delays are a little more complex but add important functionality to rules. Whilst the "Wait-For"-block is straightforward, the timer blocks should be well understood before being used - they may behave differently than expected. This chapter explains what these blocks do, sometimes displaying generated code to explain what is happening behind the scenes.
```
Comment (Confectrician) [inline configuration/rules-blockly-timers-and-delays.md:9]: Please change this to one sentence per line.
The website build will take core of a proper presentation of paragraphs.

File: configuration/rules-blockly-standard-ext.md [hunk 20]
```diff
@@ -0,0 +1,105 @@
...
+![get-hsb-of-example](images/blockly/blockly-colors-hsbof-example.png)
+
+## Lists
+
+![lists-overview](images/blockly/blockly-lists-dictionary-overview.png)
+
+### Dictionary for managing key / value pairs
+
+The dictionary is a holder for key value pairs that can be passed along as one
+Add or remove new key value pairs by clicking on the cog icon to open the popup of the dictionary block (finally click on the cog again to close the popup). Then drag the key block on the upper left into the key/value bracket on the right like depicted below:
```
Comment (Confectrician) [inline configuration/rules-blockly-standard-ext.md:82]: Please change this to one sentence per line.
The website build will take core of a proper presentation of paragraphs.

File: configuration/rules-blockly-standard-ext.md [hunk 21]
```diff
@@ -0,0 +1,105 @@
...
+
+![get-hsb-of-example](images/blockly/blockly-colors-hsbof-example.png)
+
+## Lists
+
+![lists-overview](images/blockly/blockly-lists-dictionary-overview.png)
+
+### Dictionary for managing key / value pairs
+
+The dictionary is a holder for key value pairs that can be passed along as one
```
Comment (Confectrician) [inline configuration/rules-blockly-standard-ext.md:81]:
```suggestion
The dictionary is a holder for key value pairs that can be passed along as one.
```

File: configuration/rules-blockly-standard-ext.md [hunk 22]
```diff
@@ -0,0 +1,105 @@
...
+
+HSB stands for hue-saturation-brightness which is a different way of describing a color. An RGB color can therefore be converted equally into an HSB-color.
+
+*Function:* This a special openHAB block that converts a hexadecimal color code from one of the above blocks into a String that is required when sending a color command to any item that accepts a color.
+
+- use that block and connect one of the above blocks into that one to be able to retrieve the HSB value of that block.
+
+HSB stands for hue-saturation-brightness which is a different way of describing a color. An RGB color can therefore be converted equally into an HSB-color.
+
+Function: This a special openHAB block that converts a hexadecimal color code from one of the above blocks into a String that is required when sending a color command to any item that accepts a color.
```
Comment (Confectrician) [inline configuration/rules-blockly-standard-ext.md:67]:
```suggestion
*Function:* This a special openHAB block that converts a hexadecimal color code from one of the above blocks into a String that is required when sending a color command to any item that accepts a color.
```

File: configuration/rules-blockly-standard-ext.md [hunk 23]
```diff
@@ -0,0 +1,105 @@
...
+
+![get-random-color](images/blockly/blockly-colors-random.png)
+
+*Function:* Returns a random color as a hexcode
+
+### Create RGB color
+
+![create-rgb-color](images/blockly/blockly-color-create-rgb.png)
+
+Function: Returns a color by mixing the given values in a range from 0 - 255 for reg, green and blue as a hexcode
```
Comment (Confectrician) [inline configuration/rules-blockly-standard-ext.md:44]:
```suggestion
*Function:* Returns a color by mixing the given values in a range from 0 - 255 for reg, green and blue as a hexcode
```

File: configuration/rules-blockly-standard-ext.md [hunk 24]
```diff
@@ -0,0 +1,105 @@
...
+
+{: #blockly-standard-extension-overview}
+
+## Colors
+
+The Colors section simplifies the selection of  Color Values. Technically these color RGB values are provided as a hexadecimal combination of the three values for red, green and blue (hence RGB). Each of the individual colors can range from 0 - 255 or 0-FF (hexadecimal). A color of r=255, g=255, b=255 would therefore result into #ffffff preceded by a # which identifies it as a hexadecimal number.
+
+![colors-overview](images/blockly/blockly-colors-overview.png)
+
+Note that typically RGB values are not used within openHAB for color states but instead the Color-Type expects a HSB Type. See the special hsb-conversion block below.
```
Comment (Confectrician) [inline configuration/rules-blockly-standard-ext.md:26]: Please change this to one sentence per line.
The website build will take core of a proper presentation of paragraphs.

File: configuration/rules-blockly-standard-ext.md [hunk 25]
```diff
@@ -0,0 +1,105 @@
...
+{::options toc_levels="2..4"/}
+
+- TOC
+{:toc}
+
+{: #blockly-standard-extension-overview}
+
+## Colors
+
+The Colors section simplifies the selection of  Color Values. Technically these color RGB values are provided as a hexadecimal combination of the three values for red, green and blue (hence RGB). Each of the individual colors can range from 0 - 255 or 0-FF (hexadecimal). A color of r=255, g=255, b=255 would therefore result into #ffffff preceded by a # which identifies it as a hexadecimal number.
```
Comment (Confectrician) [inline configuration/rules-blockly-standard-ext.md:22]: Please change this to one sentence per line.
The website build will take core of a proper presentation of paragraphs.

File: configuration/rules-blockly-run-and-process.md [hunk 26]
```diff
@@ -0,0 +1,163 @@
...
+Hue/Saturation
+```
+
+## Inline Script
+
+![inline-script](images/blockly/blockly-inline-script.png)
+
+*Function:* execute arbitrary ECMAScript content
+
+Provides the possibility to inject any ECMAScript at a given position. This should be used in cases when Blockly does not provide the desired functionality. Use with diligence.
```
Comment (Confectrician) [inline configuration/rules-blockly-run-and-process.md:155]: Please change this to one sentence per line.
The website build will take core of a proper presentation of paragraphs.

File: configuration/rules-blockly-run-and-process.md [hunk 27]
```diff
@@ -0,0 +1,163 @@
...
+
+### Example 1: Regex
+
+Convert  #123456 into rgb(12,34,56)
+
+![transform-regex](images/blockly/blockly-transform-example1.png)
+
+### Example 2: JsonPath transformation
+Extract the temperature from the following JSON
+```json
```
Comment (Confectrician) [inline configuration/rules-blockly-run-and-process.md:114]: Please add an empty line before the code block.

File: configuration/rules-blockly-run-and-process.md [hunk 28]
```diff
@@ -0,0 +1,163 @@
...
+{ "device": { "location": "Outside", "status": { "temperature": 23.2 }}}
+```
+by using the jsonpath
```
Comment (Confectrician) [inline configuration/rules-blockly-run-and-process.md:117]: Please ad an empty line after the code block befor starting the next paragraph.

File: configuration/rules-blockly-run-and-process.md [hunk 29]
```diff
@@ -0,0 +1,163 @@
...
+Function: Transform the given value via one of the different transformation methods
+
+### Example 1: Regex
+
+Convert  #123456 into rgb(12,34,56)
+
+![transform-regex](images/blockly/blockly-transform-example1.png)
+
+### Example 2: JsonPath transformation
+Extract the temperature from the following JSON
```
Comment (Confectrician) [inline configuration/rules-blockly-run-and-process.md:113]: Please add an empty line before the paragraph.

File: configuration/rules-blockly-run-and-process.md [hunk 30]
```diff
@@ -0,0 +1,163 @@
...
+
+### Retrieve rule context information
+
+![retrieve-context-info](images/blockly/blockly-retrieve-rule-context-info.png)
+
+*Function:* Retrieve event context related information of the rule
+
+A rule carries contextual information when triggered - this block can retrieve that information.
+
+* The rule trigger type defines which of the attributes are available to the rule - they will not all be available.
```
Comment (Confectrician) [inline configuration/rules-blockly-run-and-process.md:81]: Please use `-` for lists in all files.

Currently we have mixed usage of `-` and `*`.

File: configuration/rules-blockly-run-and-process.md [hunk 31]
```diff
@@ -0,0 +1,163 @@
...
+The  *ruleUID* can be found in the list of rules or scripts in the second line of the list (like here *helloBlockly* or *helloWorld*)
+![rule-uuid](images/blockly/blockly-run-rule-uuid.png)
+
+Parameters are optional:
+
+* If not providing parameters leave the *with context* section as is.
+* If providing parameters you **must** replace the *create empty dictionary* block with [the *dictionary* block from the Lists section](https://community.openhab.org/t/blockly-reference/128785#lists-76), modifying the number of key/value pairs to suit.
+
+Notes:
+* parameters can be retrieved in the called script via the "get context attribute"-block (see below)
```
Comment (Confectrician) [inline configuration/rules-blockly-run-and-process.md:56]: Please use `-` for lists in all files.

Currently we have mixed usage of `-` and `*`.

File: configuration/rules-blockly-run-and-process.md [hunk 32]
```diff
@@ -0,0 +1,163 @@
...
+![run-rule-or-script](images/blockly/blockly-run-rule.png)
+
+*Function:* calls another *rule* or *script* that was created via the openHAB UI.
+
+The  *ruleUID* can be found in the list of rules or scripts in the second line of the list (like here *helloBlockly* or *helloWorld*)
+![rule-uuid](images/blockly/blockly-run-rule-uuid.png)
+
+Parameters are optional:
+
+* If not providing parameters leave the *with context* section as is.
```
Comment (Confectrician) [inline configuration/rules-blockly-run-and-process.md:52]: Please use `-` for lists in all files.

Currently we have mixed usage of `-` and `*`.

File: configuration/rules-blockly-run-and-process.md [hunk 33]
```diff
@@ -0,0 +1,163 @@
...
+title: Rules Blockly - Run & Process
+---
+
+# Run & Process (Rules and Scripts)
+[return to Blockly Reference]({{base}}/configuration/rules-blockly.html)
+
+## Introduction
+
+This section contains several possibilities
+* Calling Rules or other scripts either located as files on the server or provided via the UI
```
Comment (Confectrician) [inline configuration/rules-blockly-run-and-process.md:12]: Please use `-` for lists in all files.

Currently we have mixed usage of `-` and `*`.

File: configuration/rules-blockly-items-things.md [hunk 34]
```diff
@@ -0,0 +1,181 @@
...
+```text
+thing name = nanoleaf:controller:645E3A484A83
+```
+
+### Get Thing Status
+
+![blockly-getThingStatus](images/blockly/blockly-getThingStatus.png)
+Function: Gets a **Thing Status** for use in other Thing related functions
+
+* Clicking 'MyThing' displays a list of **Things** to pick from.
```
Comment (Confectrician) [inline configuration/rules-blockly-items-things.md:168]: Please use `-` for lists in all files.

Currently we have mixed usage of `-` and `*`.

File: configuration/rules-blockly-items-things.md [hunk 35]
```diff
@@ -0,0 +1,181 @@
...
+
+## Things
+
+### Thing
+
+![blockly-thing](images/blockly/blockly-thing.png)
+
+Function: Retrieves a specific **Thing** for use in other thing related functions.
+
+* Clicking 'MyThing' displays a list of **Things** to pick from
```
Comment (Confectrician) [inline configuration/rules-blockly-items-things.md:149]: Please use `-` for lists in all files.

Currently we have mixed usage of `-` and `*`.

File: configuration/rules-blockly-items-things.md [hunk 36]
```diff
@@ -0,0 +1,181 @@
...
+* name: String
+* label: String
+* state: State
+* category: String
+* tag: Array, e.g.
+```
+[plannedTimes]
+```
+* groups: Array, e.g.
+```
```
Comment (Confectrician) [inline configuration/rules-blockly-items-things.md:98]: Please add an empty line before the code block.

File: configuration/rules-blockly-items-things.md [hunk 37]
```diff
@@ -0,0 +1,181 @@
...
+Note: currently a String is not always returned. [See this thread](https://community.openhab.org/t/blockly-cannot-split-text-string-from-string-item/130819) for reference and potential workaround.
+
+### Get Members of Group
+![blockly-getMembers](images/blockly/blockly-getMembers.png)
+
+Function: Gets the members of a **group**
+
+* returns a collection of items which should be used with a for-each-block to loop over the items
+* it can be attached to a log-block which would list all items in that block in the form a string representation as follows
+````
```
Comment (Confectrician) [inline configuration/rules-blockly-items-things.md:73]: Please add an empty line before the code block.

File: configuration/rules-blockly-items-things.md [hunk 38]
```diff
@@ -0,0 +1,181 @@
...
+
+> ![items-and-things](images/blockly/blockly-items-and-things.png)
+
+
+### Item and Thing Blocks
+Most of the time you will want to get and set the state of an *item* - set a switch to ON, or get a temperature. Sometimes you may want to access the *thing* directly. Both are possible using the item-block and thing-block
+
+> ![blockly-item-thing](images/blockly/blockly-item-thing.png)
+
+However, these blocks are not useful by themselves - they are always used together with another block. The example below uses the *get state of item* block to retrieve the *MainSwitch* *item* state, before checking if that is equal to ON within a standard comparison block
```
Comment (Confectrician) [inline configuration/rules-blockly-items-things.md:28]: Please change this to one sentence per line.
The website build will take core of a proper presentation of paragraphs.

File: configuration/rules-blockly-items-things.md [hunk 39]
```diff
@@ -0,0 +1,181 @@
...
+
+{: #blockly-items-and-things-overview}
+
+## Overview of the Items and Things category
+
+> ![items-and-things](images/blockly/blockly-items-and-things.png)
+
+
+### Item and Thing Blocks
+Most of the time you will want to get and set the state of an *item* - set a switch to ON, or get a temperature. Sometimes you may want to access the *thing* directly. Both are possible using the item-block and thing-block
```
Comment (Confectrician) [inline configuration/rules-blockly-items-things.md:24]: Please change this to one sentence per line.
The website build will take core of a proper presentation of paragraphs.

File: configuration/rules-blockly-items-things.md [hunk 40]
```diff
@@ -0,0 +1,181 @@
+---
+layout: documentation
+title: Rules Blockly - Items & Things
+---
+
+# Items &  Things
+[return to Blockly Reference]({{base}}/configuration/rules-blockly.html)
+
+*Items* and *Things* are the [major entities of openHAB](https://www.openhab.org/docs/concepts/) to control and monitor the home. These can be accessed via the "Items & Things" section of the [Blockly Toolbox]({{base}}/configuration/rules-blockly.html#blockly-toolbox).
```
Comment (Confectrician) [inline configuration/rules-blockly-items-things.md:9]: Please change this to one sentence per line.
The website build will take core of a proper presentation of paragraphs.

**Review by Confectrician — COMMENTED** [review]
This is a first look.
Don't get a heart attack for the many comments. (Sorry. 😄 )

Mostly the topics are similar.

- One sentence per line
- Empty lines around code blocks, paragraphs and headings
- Using the same list style for all lists

We use one line per sentence to keep track of changes in git.
The website build can recognize paragraphs, headings and so on with the empty lines as separator.

I checked most of teh images already and many of the files for basic markdown syntax problems.

If you are using vscode you could also install a markdown linter extension.
I am using this too, with the same markdown rules applied as our autoamded check here does.
I can provide more information if its from interest for you.

**Comment by stefan-hoehn** [comment]
> Yes it contains the whole documentation to 3.2 and even 3.3 already

**Review by Confectrician — APPROVED** [review]
Thanks for your efforts @stefan-hoehn 👍

The last error is not related with this pull request, so we can go on.
A big progress for the blockly docs. :)

**Comment by hmerk** [comment]
@stefan-hoehn @Confectrician 
just noticed that „Value Storage“, „Run & Process“ and „Logging“ show the same picture. Is this intended ?

**Comment by Confectrician** [comment]
To be honest, I didn't check every image link because it were pretty much and I am not that deep into Blockly.

I concentrated more on the markdown appearance.

Let's wait what Stefan says.

**Pull request merged**